Source/WebCore/rendering/GlyphDisplayListCache.cpp

```diff
@@ -117,19 +119,28 @@ RefPtr<const DisplayList::DisplayList> GlyphDisplayListCache::getDisplayList(con
             return nullptr;
     }
 
-    if (auto iterator = m_entries.find<GlyphDisplayListCacheKeyTranslator>(GlyphDisplayListCacheKey { textRun, font, context }); iterator != m_entries.end()) {
-        Ref entry { iterator->get() };
-        RefPtr result = &entry->displayList();
-        const_cast<LayoutRun&>(run).setIsInGlyphDisplayListCache();
-        m_entriesForLayoutRun.add(&run, WTFMove(entry));
-        return result;
+    if (auto iterator = m_entries.findIf([&](auto& weakEntry) {
+        return GlyphDisplayListCacheKeyTranslator::equal(weakEntry, GlyphDisplayListCacheKey { textRun, font, context });
+    }); iterator != m_entries.end()) {
+        RefPtr<GlyphDisplayListCacheEntry> entry = iterator->get();
+        if (!entry)
+            m_entries.remove(iterator);
+        else {
+            const_cast<LayoutRun&>(run).setIsInGlyphDisplayListCache();
+            RefPtr result = &entry->displayList();
+            m_entriesForLayoutRun.add(&run, entry.releaseNonNull());
+            return result;
+        }
     }
 
     if (RefPtr displayList = font.displayListForTextRun(context, textRun)) {
         Ref entry = GlyphDisplayListCacheEntry::create(displayList.releaseNonNull(), textRun, font, context);
         Ref result = entry->displayList();
-        if (canShareDisplayList(result))
-            m_entries.add(entry.get());
+        if (canShareDisplayList(result)) {
+            m_entries.append(entry.get());
```

Comment:
since 20 is such a low number, it might make sense to just reserveCapacity(s_max... + 1) here, so either we spend 0 memory or we spend max amount of entries.
Doesn't really make a difference, so might not be worth resubmitting. Just an observation.